Mathlib/Topology/Instances/ENat.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2022 Peter Nelson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Nelson
+-/
+import Mathlib.Data.ENat.Basic
+import Mathlib.Topology.Algebra.Monoid
+import Mathlib.Topology.Instances.Discrete
+
+/-!
+# Topology on extended natural numbers
+-/
+
+open Set
+
+open scoped Topology Filter
+
+namespace ENat
+
+instance : TopologicalSpace ℕ∞ := Preorder.topology ℕ∞
+
+instance : OrderTopology ℕ∞ := ⟨rfl⟩
+
+@[simp] theorem range_natCast : range ((↑) : ℕ → ℕ∞) = Iio ⊤ :=
+  WithTop.range_coe
```

Comment:
Does `Data.ENat.Basic` have all `import`s for this lemma? It isn't specific to topology.

Reply:
Answer: no, let's leave here for now.

Reply:
#15774 has a version of this lemma in `ENat.Basic`